homeassistant/components/wilight/switch.py

```diff
@@ -0,0 +1,393 @@
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if device is on."""
+        return self._status.get("on", False)
+
+    @property
+    def watering_time(self) -> int | None:
+        """Return watering time of valve switch.
+
+        None is unknown, 1 is minimum, 1800 is maximum.
+        """
+        return self._status.get("timer_target")
+
+    @property
+    def trigger_1(self) -> str | None:
+        """Return trigger_1 of valve switch."""
+        return self._status.get("trigger_1")
+
+    @property
+    def trigger_2(self) -> str | None:
+        """Return trigger_2 of valve switch."""
+        return self._status.get("trigger_2")
+
+    @property
+    def trigger_3(self) -> str | None:
+        """Return trigger_3 of valve switch."""
+        return self._status.get("trigger_3")
+
+    @property
+    def trigger_4(self) -> str | None:
+        """Return trigger_4 of valve switch."""
+        return self._status.get("trigger_4")
+
+    @property
+    def trigger_1_description(self) -> str | None:
+        """Return trigger_1_description of valve switch."""
+        return wilight_to_hass_trigger(self._status.get("trigger_1"))
+
+    @property
+    def trigger_2_description(self) -> str | None:
+        """Return trigger_2_description of valve switch."""
+        return wilight_to_hass_trigger(self._status.get("trigger_2"))
+
+    @property
+    def trigger_3_description(self) -> str | None:
+        """Return trigger_3_description of valve switch."""
+        return wilight_to_hass_trigger(self._status.get("trigger_3"))
+
+    @property
+    def trigger_4_description(self) -> str | None:
+        """Return trigger_4_description of valve switch."""
+        return wilight_to_hass_trigger(self._status.get("trigger_4"))
+
+    @property
+    def state_attributes(self) -> dict[str, Any]:
+        """Return the state attributes."""
+        attr: dict[str, Any] = {}
+
+        if self.watering_time is not None:
+            attr[ATTR_WATERING_TIME] = self.watering_time
+
+        if self.trigger_1 is not None:
+            attr[ATTR_TRIGGER_1] = self.trigger_1
+
+        if self.trigger_2 is not None:
+            attr[ATTR_TRIGGER_2] = self.trigger_2
+
+        if self.trigger_3 is not None:
+            attr[ATTR_TRIGGER_3] = self.trigger_3
+
+        if self.trigger_4 is not None:
+            attr[ATTR_TRIGGER_4] = self.trigger_4
+
+        if self.trigger_1_description is not None:
+            attr[ATTR_TRIGGER_1_DESC] = self.trigger_1_description
+
+        if self.trigger_2_description is not None:
+            attr[ATTR_TRIGGER_2_DESC] = self.trigger_2_description
+
+        if self.trigger_3_description is not None:
+            attr[ATTR_TRIGGER_3_DESC] = self.trigger_3_description
+
+        if self.trigger_4_description is not None:
+            attr[ATTR_TRIGGER_4_DESC] = self.trigger_4_description
+
+        return attr
+
+    @property
+    def supported_features(self) -> int:
```

Comment:
`supported_features` is not a property of the `switch` platform.  Please use another name for this

Reply:
Correct, I will not need this property and I will delete it...